modules/images/image-loading-optimization/class-ilo-html-tag-processor.php

```diff
@@ -183,7 +183,11 @@ public function open_tags(): Generator {
 				yield $tag_name;
 
 				// Immediately pop off self-closing tags.
-				if ( in_array( $tag_name, self::VOID_TAGS, true ) ) {
+				if (
+					in_array( $tag_name, self::VOID_TAGS, true )
+					||
+					( $p->has_self_closing_flag() && $this->is_foreign_element() )
+				) {
 					array_pop( $this->open_stack_tags );
 				}
```

Comment:
@dmsnell: FYI, I'm pretty happy with how `has_self_closing_flag` and the foreign element aspect of the HTML spec simplified things! Now it can properly get breadcrumbs/XPaths for each SVG/MathML element as well.